src/avro/src/schema.rs

```diff
@@ -579,18 +580,52 @@ impl fmt::Debug for FullName {
 pub type Documentation = Option<String>;
 
 impl Name {
-    /// Create a new `Name`.
-    /// No `namespace` nor `aliases` will be defined.
-    pub fn new(name: &str) -> Name {
-        Name {
-            name: name.to_owned(),
-            namespace: None,
-            aliases: None,
+    /// Reports whether the given string is a valid Avro name.
+    pub fn is_valid(name: &str) -> bool {
+        static MATCHER: Lazy<Regex> =
+            Lazy::new(|| Regex::new(r"(^[A-Za-z_][A-Za-z0-9_]*)$").unwrap());
```

Comment:
if there is avro docs that describe this regex, it would be great to link to them here!

Reply:
Done.